codec/src/main/java/io/netty/handler/codec/compression/JZlibDecoder.java

```diff
@@ -129,29 +152,43 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
                     }
 
                     switch (resultCode) {
-                    case JZlib.Z_NEED_DICT:
-                        if (dictionary == null) {
-                            ZlibUtil.fail(z, "decompression failure", resultCode);
-                        } else {
-                            resultCode = z.inflateSetDictionary(dictionary, dictionary.length);
-                            if (resultCode != JZlib.Z_OK) {
-                                ZlibUtil.fail(z, "failed to set the dictionary", resultCode);
+                        case JZlib.Z_NEED_DICT:
+                            if (dictionary == null) {
+                                ZlibUtil.fail(z, "decompression failure", resultCode);
+                            } else {
+                                resultCode = z.inflateSetDictionary(dictionary, dictionary.length);
+                                if (resultCode != JZlib.Z_OK) {
+                                    ZlibUtil.fail(z, "failed to set the dictionary", resultCode);
+                                }
+                            }
+                            break;
+                        case JZlib.Z_STREAM_END:
+                            z.inflateEnd();
```

Comment:
I couldn't find any javadocs on this method...can you provide justification why this should be called unconditionally?